js/components/codeScanner.js

```diff
@@ -0,0 +1,68 @@
+Fliplet.FormBuilder.field('codeScanner', {
+  name: 'Code scanner',
+  category: 'Advanced',
+  props: {
+    value: {
+      type: String
+    },
+    description: {
+      type: String
+    },
+    placeholder: {
+      type: String
+    }
+  },
+  data: function() {
+    return {
+      errorMessage: null
+    };
+  },
+  validations: function() {
+    var rules = {
+      value: {}
+    };
+
+    if (this.required && !this.readonly) {
+      rules.value.required = window.validators.required;
+    }
+
+    return rules;
+  },
+  created: function() {
+    Fliplet.FormBuilder.on('reset', this.onReset);
+  },
+  destroyed: function() {
+    Fliplet.FormBuilder.off('reset', this.onReset);
+  },
+  methods: {
+    openScanner: function() {
+      var $vm = this;
+
+      if (this.readonly) {
+        return;
+      }
+
+      Fliplet.Barcode.scan().then(function(result) {
+        if (result.text) {
+          $vm.value = result.text;
+        }
+      })
+        .catch(function(error) {
+          $vm.errorMessage = error;
```

Comment:
The error may not be a string.

```suggestion
          $vm.errorMessage = Fliplet.parseError(error);
```